cmd/wallet-web/src/router/index.js

```diff
@@ -71,7 +71,30 @@ router.beforeEach(async (to, from) => {
       console.error('error authenticating user: invalid hash received');
       return false;
     }
-  } else if (to.matched.some((record) => record.meta.requiresAuth)) {
+  }
+
+  if (to.matched.some((record) => record.meta.requiresCHAPI)) {
+    // Get enableCHAPI from wallet config
+    if (store.getters.getEnableCHAPI === undefined) {
+      if (store.dispatch('loadUser') && store.getters.getEnableCHAPI === undefined) {
+        await store.dispatch('initOpts');
+      }
+    }
```

Comment:
Can we simplify this block to something like below?

>     if (store.getters.getEnableCHAPI === undefined && store.dispatch('loadUser')) {
>        await store.dispatch('initOpts');
>      }

Reply:
The logic behind is to firstly try to get the value from Vuex, secondly fetch it from the local storage, and lastly make an axios call. As soon as the value is not `undefined`, we won't make any further steps. There are edge cases, when the value is not available in vuex or local storage, so we use api call as a last resort.
@rolsonquadras @sudeshrshetty